openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns.go

```diff
@@ -142,3 +143,32 @@ func validateTolerations(versionedTolerations []corev1.Toleration, fldPath *fiel
 	allErrors = append(allErrors, apivalidation.ValidateTolerations(unversionedTolerations, fldPath)...)
 	return allErrors
 }
+
+// validateUpstreamResolvers validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolvers(upstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tc := operatorv1.DNSTransportConfig{}
+
+	// TransportConfig is empty so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig == tc {
+		return errs
+	}
+
+	// Transport is empty or Cleartext so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig.Transport == "" || upstreamResolvers.TransportConfig.Transport == operatorv1.CleartextTransport {
+		return errs
+	}
+
+	// Transport is TLS so no mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig.Transport == operatorv1.TLSTransport {
+		for i, upstream := range upstreamResolvers.Upstreams {
+			errPath := fieldPath.Child("upstreams")
+			if upstream.Type == operatorv1.SystemResolveConfType {
+				errMessage := fmt.Sprintf("upstream %s has type %s but %s has been configured", upstream.Address, upstream.Type, operatorv1.TLSTransport)
```

Comment:
`upstream.Address` _should_ be empty if `upstream.Type` is `operatorv1.SystemResolveConfType`.  Anyway, the error message has the index, which is sufficient to identify the problematic field.  
```suggestion
				errMessage := fmt.Sprintf("upstream has type %s but %s has been configured", upstream.Type, operatorv1.TLSTransport)
```

Reply:
Fixed locally!